prov/efa/src/efa.h

```diff
@@ -106,6 +106,7 @@ struct efa_fabric {
 struct efa_ep_addr {
 	uint8_t			raw[16];
 	uint16_t		qpn;
+	struct efa_ep_addr	*next;
```

Comment:
Is this exposed to the app in any way?  It's odd to have a pointer as part of an address.

Reply:
efa_ep_addr is used internally, not exposed to app. We wanted to re-use the efa_ep_addr structure to handle multi-EFA case, so added a pointer there.

I have two concerns on shm provider. 

1. CMA requires ptrace permission on Ubuntu 16.04 and 18.04 to work. So shm would fail if the users do not want to compromise their systems. How can we handle this case? Is there a fallback solution?

2. I was having issue when using fi_cq_readfrom with shm provider. It seems addr does not have assignment (https://github.com/ofiwg/libfabric/blob/master/prov/shm/src/smr_progress.c#L331). And the returned addr of fi_cq_readfrom is (uint32_t) (uintptr_t) addr instead of fi_addr_t.

Reply:
There isn't a fallback solution in place.  For medium sized messages, a segmentation protocol through bounce buffers might work.  Larger messages may require mapping in the buffers.

I don't understand the fi_addr_t issue.

Reply:
I think I understand the fi_addr_t issue. Drafting up a patch for it now.

Reply:
See PR #5329 for fix to the addr issue

Reply:
PR #5329 solved the addr issue I had earlier.